ansible_ai_connect/ai/api/views.py

```diff
@@ -820,195 +823,67 @@ class GenerationPlaybook(APIView):
-            )
-            raise WcaRequestIdCorrelationFailureException(cause=e)
-
-        except WcaEmptyResponse as e:
-            exception = e
-            logger.exception(
-                f"WCA returned an empty response for playbook generation {generation_id}"
-            )
-            raise WcaEmptyResponseException(cause=e)
-
-        except WcaCloudflareRejection as e:
-            exception = e
-            logger.exception(
-                f"Cloudflare rejected the request for playbook generation {generation_id}"
-            )
-            raise WcaCloudflareRejectionException(cause=e)
-
-        except WcaHAPFilterRejection as e:
-            exception = e
-            logger.exception(
-                f"WCA Hate, Abuse, and Profanity filter rejected "
-                f"the request for playbook generation {generation_id}"
-            )
-            raise WcaHAPFilterRejectionException(cause=e)
+        request_serializer = GenerationPlaybookRequestSerializer(data=request.data)
+        request_serializer.is_valid(raise_exception=True)
+        generation_id = str(request_serializer.validated_data.get("generationId", ""))
+        create_outline = request_serializer.validated_data["createOutline"]
+        outline = str(request_serializer.validated_data.get("outline", ""))
+        text = request_serializer.validated_data["text"]
+        custom_prompt = str(request_serializer.validated_data.get("customPrompt", ""))
+        wizard_id = str(request_serializer.validated_data.get("wizardId", ""))
+        self.req_model_id = str(request_serializer.validated_data.get("model", ""))
 
-        except WcaUserTrialExpired as e:
-            exception = e
-            logger.exception(
-                f"User trial expired, when requesting playbook generation {generation_id}"
-            )
-            raise WcaUserTrialExpiredException(cause=e)
+        self.event.generationId = generation_id
+        self.event.wizardId = wizard_id
+        self.event.modelName = model_id
 
-        except WcaInstanceDeleted as e:
-            exception = e
-            logger.exception(
-                "WCA Instance has been deleted when requesting playbook generation "
-                f"{generation_id} for model {e.model_id}"
+        self.event.create_outline = create_outline
+        llm: ModelPipelinePlaybookGeneration = apps.get_app_config("ai").get_model_pipeline(
+            ModelPipelinePlaybookGeneration
+        )
+        playbook, outline, warnings = llm.invoke(
+            PlaybookGenerationParameters.init(
+                request=request,
+                text=text,
+                custom_prompt=custom_prompt,
+                create_outline=create_outline,
+                outline=outline,
+                generation_id=generation_id,
+                model_id=model_id,
             )
-            raise WcaInstanceDeletedException(cause=e)
+        )
+        self.event.playbook_length = len(playbook)
```

Comment:
Should this be moved lower down and use `self.event.playbook_length = len(anonymized_playbook)`?

The event [used to contain](https://github.com/ansible/ansible-ai-connect-service/pull/1485/files#diff-ecfb6919dfd8379aafba96af7457b253e4dce528897dfe6bfc207ca2b3b2ada9L969) the length of the anonymized playbook.